specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+or not ready.
+
+The Trigger's controller MUST also set the `status.subscriberUri` field based on
+resolving the `spec.subscriber` field before setting the `Ready` condition to
+`true`. If the `spec.subscriber.ref` field points to a resource which does not
+exist or cannot be resolved via
+[Addressable resolution](#addressable-resolution), the Trigger MUST set the
+`Ready` condition to `false`, and at least one condition MUST indicate the
+reason for the error. The Trigger SHOULD also set `status.subscriberUri` to the
+empty string if the `spec.subscriber.ref` cannot be resolved.
+
+If the Trigger's `spec.delivery.deadLetterSink` field it set, it MUST be
+resolved to a URI and reported in `status.deadLetterSinkUri` in the same manner
+as the `spec.subscriber` field before setting the `Ready` condition to `true`.
+
+Once created, the Trigger's `spec.broker` MUST NOT permit updates; to change the
+`spec.broker`, the Trigger can instead be deleted and re-created. This pattern
+is chosen to make it clear that changing the `spec.broker` is not an atomic
+operation, as it could span multiple storage systems. Changes to
+`spec.subscriber`, `spec.filter` and other fields SHOULD be permitted, as these
+could occur within a single storage system.
+
+When a Trigger becomes associated with a Broker (either due to creating the
+Trigger or the Broker), the Trigger MUST only set the `Ready` condition to
+`true` after the Broker has been configured to send all future events matching
+the `spec.filter` to the Trigger's `spec.subscriber`. The Broker MAY send some
+events to the Trigger's `spec.subscriber` prior to the Trigger's
+`Ready`condition being set to `true`. When a Trigger is deleted, the Broker MAY
+send some additional events to the Trigger's `spec.subscriber` after the
+deletion.
+
+### Channel Lifecycle
+
+A Channel represents an Addressable endpoint (i.e. it MUST have a
+`status.address.url` field) which can receive, store, and forward events to
+multiple recipients (Subscriptions). Subscriptions MUST be associated with a
+Channel based on the `spec.channel` field on the Subscription; it is expected
+that the controller for a Channel will also control the associated
+Subscriptions. When the Channel's `Ready` condition is `true`, the Channel MUST
+provide a `status.address.url` which accepts all valid CloudEvents and MUST
+attempt to forward the received events to each associated Subscription whose
+`Ready` condition is `true`. As described in the
+[Subscription Lifecycle](#subscription-lifecycle) section, a Channel MAY forward
+events to an associated Subscription which does not currently have a `true`
+`Ready` condition, including events received by the Channel before the
+`Subscription` was created.
+
+When a Channel is created, its `spec.channelTemplate` field MUST be populated to
+indicate which of several possible Channel implementations to use. It is
+RECOMMENDED to default the `spec.channelTemplate` field on creation if it is
```

Comment:
Like above, this RECOMMENDED contradicts the "MUST be populated".

Reply:
But like my previous comment, I'd like it so it's clear what the client vs server is required to do

Reply:
Now that I realized we've merged client and server requirements, as I'm reading more it really feels like we need to be much more clear throughout all of these docs about who these MUSTs apply to. It might help if the specs were split (either into separate docs, or separate sections) where one is for clients and one is for servers. Then it someone wants to only implement one side of the interaction they'll have everything they need in one spot - and not have to read through the other side's job just to make sure they don't miss something. Maybe that kind of split is too much work, but I do think we need to split it even at the paragraph level.